tests/source/enum.rs

```diff
@@ -125,3 +125,16 @@ enum Loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo
 enum Looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong {}
 enum Loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong {}
 enum Loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong { Foo }
+
+// #1046
+pub enum Entry<'a, K: 'a, V: 'a> {
+    // This attribute should stay on the same line.
+    Vacant(
+        #[ stable( feature = "rust1", since = "1.0.0" ) ]   VacantEntry<'a, K, V>,
+    ),
+    // This attribute should be kept on the previous line.
```

Comment:
Sorry for the late reply, I missed the notification of your comment.

> As a strawman rule, I suggest we could put attributes on the same line as fields if the whole struct/variant is on one line, and put them on their own line if we would otherwise be multi-line. How does that sound?

I like this, thanks!